core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1303,19 +1330,14 @@ default Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         return Optional.empty();
     }
 
-    /**
-     * Returns a table handle representing a versioned table. A versioned table differs by having an additional specifier for version.
-     */
-    default ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName, Optional<ConnectorTableVersion> startVersion, Optional<ConnectorTableVersion> endVersion)
-    {
-        throw new TrinoException(NOT_SUPPORTED, "This connector does not support versioned tables");
-    }
-
     /**
      * Returns whether a specified version type is supported by the connector for a given travel type and table name
      */
     default boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, Type versioning)
```

Comment:
Should this method be actually _table_ specific or rather connector specific?

https://github.com/trinodb/trino/pull/10258/files#diff-e1cb17efec6787989f9df9ee40c4f2809ff3fe946cd2ec721ff8932b131997b8R328

Iceberg doesn't seem to make use of the `SchemaTableName` argument.
Any idea in which scenario would the `SchemaTableName` argument be necessary?

@findepi  @alexjo2144

Reply:
That's probably a question to #8773 which added this method in this shape.
I guess it may be useful for a connector that has different backends/storages/layouts. Like MySQL with different "engines".

Reply:
> Should this method be actually _table_ specific or rather connector specific?
> 
> https://github.com/trinodb/trino/pull/10258/files#diff-e1cb17efec6787989f9df9ee40c4f2809ff3fe946cd2ec721ff8932b131997b8R328
> 
> Iceberg doesn't seem to make use of the `SchemaTableName` argument. Any idea in which scenario would the `SchemaTableName` argument be necessary?
> 
> @findepi @alexjo2144

If I remember correctly, Iceberg uses manifest files to support versioning, and users have control of how  many transactions to store. Can users specify that only a single transaction can be stored? If so, that might be a situation where time travel is disabled for a table, although ultimately I just don't think it makes sense in the Iceberg specific context.

The SPIs are designed for general implementation and we designed it to give connectors the flexibility to take table specific metadata or details into account.

Reply:
> Can users specify that only a single transaction can be stored?  If so, that might be a situation where time travel is disabled for a table, although ultimately I just don't think it makes sense in the Iceberg specific context.

I don't think it's possible. Operations that create new snapshots are separate from old snapshot expiration.

While it would be possible to implement a single operation that does both, it probably wouldn't make sense.
And even then, table still can be reported as supporting versioning, just having a rather short history.

> The SPIs are designed for general implementation and we designed it to give connectors the flexibility to take table specific metadata or details into account.

as above (https://github.com/trinodb/trino/pull/12542#discussion_r882110623): yes, it makes sense